src/main/java/com/pngencoder/PngEncoder.java

```diff
@@ -38,38 +57,88 @@ public PngEncoder() {
     public PngEncoder withBufferedImage(BufferedImage bufferedImage) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code compressionLevel}.
+     * The new PngEncoder will use the provided {@code compressionLevel}.
+     *
+     * @param compressionLevel input image (must be between -1 and 9 inclusive)
+     * @return a new PngEncoder
+     */
     public PngEncoder withCompressionLevel(int compressionLevel) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code multiThreadedCompressionEnabled}.
+     * The new PngEncoder will use the provided {@code multiThreadedCompressionEnabled}.
+     *
+     * @param multiThreadedCompressionEnabled when {@code true}, multithreaded compression will be used
+     * @return a new PngEncoder
+     */
     public PngEncoder withMultiThreadedCompressionEnabled(boolean multiThreadedCompressionEnabled) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns a new PngEncoder which has the same configuration as this one except {@code srgbRenderingIntent}.
+     * The new PngEncoder will use the provided {@code srgbRenderingIntent}.
+     *
+     * @param srgbRenderingIntent todo
+     * @return a new PngEncoder
+     */
     public PngEncoder withSrgbRenderingIntent(PngEncoderSrgbRenderingIntent srgbRenderingIntent) {
         return new PngEncoder(bufferedImage, compressionLevel, multiThreadedCompressionEnabled, srgbRenderingIntent);
     }
 
+    /**
+     * Returns the {@code BufferedImage} provided at construction time.
+     * @return the {@code BufferedImage} provided at construction time
+     */
     public BufferedImage getBufferedImage() {
         return bufferedImage;
     }
 
+    /**
+     * Returns the compression level provided at construction time.
+     * @return the compression level provided at construction time
+     */
     public int getCompressionLevel() {
         return compressionLevel;
     }
 
+    /**
+     * Returns the multiThreadedCompressionEnabled provided at construction time.
+     * @return the multiThreadedCompressionEnabled provided at construction time
+     */
     public boolean isMultiThreadedCompressionEnabled() {
         return multiThreadedCompressionEnabled;
     }
 
+    /**
+     * Returns the {@code PngEncoderSrgbRenderingIntent} provided at construction time.
+     * @return the {@code PngEncoderSrgbRenderingIntent} provided at construction time
+     */
     public PngEncoderSrgbRenderingIntent getSrgbRenderingIntent() {
         return srgbRenderingIntent;
     }
 
+    /**
+     * Encodes PngEncoder to outputStream.
```

Comment:
"Encode PngEncoder ..." feels wrong.
It is the image we are encoding, not the encoder itself.

Perhaps "Encodes the image to outputStream".

The same comment applies to the different "to..." methods below.